### DISCUSSION TITLE
Disallow certain packages on the client

### DISCUSSION THREAD

**Discussion by i124q2n8** [remix-run/remix#9678] [Proposals]
I've read most of the remix docs including [this](https://remix.run/docs/en/main/guides/constraints) page. Maybe I've missed something, but it seems that there is no option to disallow certain packages explicitly to be bundle in the client bundle. Is this something that is planned for the future or is there another way to do this?

Why would you want this:

It sometime happens that you include packages in the client build by mistake and have a hard time finding them. This is because not all packages will break the build or have any direct effect on the page. But they certainly increase the bundle size and load time.
An example of a package that only affect the bundle size would be zod/typebox, because it does work in the browser. But in most cases you only want use it on the server to validate data received via actions and loaders.

Explicitly disallowing (or allowing) packages would be a nice dx improvement. This way you could disallow all packages that should never be included in the client bundle and immediately get a warning (with the file that imported it) if you include them for some reason.

**Comment by gustavopch** [comment]
This is what you want: https://github.com/pcattori/vite-env-only
